doc/developer/translating-enums.md

```diff
@@ -0,0 +1,278 @@
+ETT__ArgType_C3 (as : sort 0) : sort 0 = TupleR (ETT__ArgType_C3_LS as);
+```
+
+Now we can put the above representations into a list of types (`ListSort`):
+
+```
+ETT__LS : sort 0 -> ListSort;
+ETT__LS as = LS_Cons    (ETT__ArgType_C1 as)
+              (LS_Cons  (ETT__ArgType_C2 as)
+               (LS_Cons (ETT__ArgType_C3 as)
+                 LS_Nil));
+```
+
+Now, we can define the SAWCore representation of `ETT` as a
+polymorphic Sum of Products thus:
+
+```
+ETT : (as : sort 0) -> sort 0;
+ETT as = EithersV (ETT__LS as);
+```
+
+### Constructing Constructors
+
+`Left` and `Right` (with type `Either`) are in the SAWCore prelude,
+but it's still very tedious creating all the type arguments to pass.
+Note the structure of our three constructors:
+
+```
+C1 : (as : sort 0) -> listSortGet (ETT__LS as) 0 -> ETT as;
+C1 as x = Left (listSortGet (ETT__LS as) 0) (EithersV (listSortDrop (ETT__LS as) 1))
+               x;
+
+C2 : (as : sort 0) -> listSortGet (ETT__LS as) 1 -> ETT as;
+C2 as x =
+  Right (listSortGet (ETT__LS as) 0) (EithersV (listSortDrop (ETT__LS as) 1))
+  (Left (listSortGet (ETT__LS as) 1) (EithersV (listSortDrop (ETT__LS as) 2))
+   x);
+
+C3 : (as : sort 0) -> listSortGet (ETT__LS as) 2 -> ETT as;
+C3 as x =
+ Right   (listSortGet (ETT__LS as) 0) (EithersV (listSortDrop (ETT__LS as) 1))
+  (Right (listSortGet (ETT__LS as) 1) (EithersV (listSortDrop (ETT__LS as) 2))
+   (Left (listSortGet (ETT__LS as) 2) (EithersV (listSortDrop (ETT__LS as) 3))
+   x));
+```
+
+### Constructing `case` for `ETT`
+
+Here is the definition of our function that implements `case` for `ETT`:
+
+```
+ETT_case  : (as : sort 0)
+         -> (b: sort 0)
+         -> (arrowsType (ETT__ArgType_C1_LS as) b)
+         -> (arrowsType (ETT__ArgType_C2_LS as) b)
+         -> (arrowsType (ETT__ArgType_C3_LS as) b)
+         -> ETT as
+         -> b;
+
+ETT_case as b f1 f2 f3 =
+  eithersV b
+    (FunsTo_Cons b (ETT__ArgType_C1 as) (\(x: ETT__ArgType_C1 as) -> f1)
+    (FunsTo_Cons b (ETT__ArgType_C2 as) (\(x: ETT__ArgType_C2 as) -> f2 x.1)
+    (FunsTo_Cons b (ETT__ArgType_C3 as) (\(x: ETT__ArgType_C3 as) -> f3 x.1 x.2)
+    (FunsTo_Nil b))));
+```
+
+See the prelude for details of `FunsTo(FunsTo_Cons,FunsTo_Nil)`.  But
+hopefully this is generally clear from the above defininition.
+Note that this definition is polymorphic on `b`, the result of the
+case expression, as well as on `as` the type arguments of `ETT`.
+
+### Translating Crytpol `case` expressions to SAWCore `ETT_case` applications
+
+For a given case in an Expression,
+```
+  case scrutinee  of
+     C1     -> RHS1
+     C2 ... -> RHS2
+     C3 ... -> RHS3
+```
+We create an application of `ETT_case` to the proper arguments.
+See the code for the gory details, but a couple key things to note are
+  * We must determine the type of the `scrutinee`: `ETT T1` 
+  * We must determine the type of the whole expression, that is `B`
+This allows us to apply the type arguments for `ETT_case` giving this
```

Comment:
Put a newline between these two lines, or GitHub will render the markdown as though "This allows us to apply [...]" is part of the same paragraph as the bullet point:

```suggestion
  * We must determine the type of the whole expression, that is `B`

This allows us to apply the type arguments for `ETT_case` giving this
```